pkg/sink/sink.go

```diff
@@ -195,24 +196,30 @@ func (r Sink) processTrigger(t *triggersv1.EventListenerTrigger, request *http.R
 	return nil
 }
 
-func (r Sink) ExecuteInterceptors(t *triggersv1.EventListenerTrigger, in *http.Request, event []byte, log *zap.SugaredLogger) ([]byte, http.Header, error) {
+// ExecuteInterceptor executes all interceptors for the Trigger and returns back the body, header, and extensions to use.
+// When TEP-0026 is fully implemented, this function will only return the added extensions.
+func (r Sink) ExecuteInterceptors(t *triggersv1.EventListenerTrigger, in *http.Request, event []byte, log *zap.SugaredLogger, eventID string) ([]byte, http.Header, map[string]interface{}, error) {
```

Comment:
We might want to return `([]byte, http.Header, *InterceptorResponse, error)` so that we can distinguish "we failed to execute the interceptor" vs "the interceptor returned an error, but we processed it correctly" errors.

Reply:
We can differentiate that based on error also but InterceptorResponse along with Error is probably better. Do we need to document this behaviour so that other interceptor also return InterceptorResponse like this?

Reply:
done.